File: old_cf_python/cf_1999e.py
def inp():
    return(int(input()))
def inlt():
    return(list(map(int,input().split())))
def insr():
    s = input()
    return(list(s[:len(s)]))
def invr():
    return(map(int,input().split()))

def solve(pref, cnt):

    l, r = invr()
    r += 1
    sum = pref[r] - pref[l]
    mn = cnt[l]

    print(sum + mn)

N = 200_007
pref = [0] * N
cnt = [0] * N

def main():
   
    for i in range(1, N - 1):
        cnt[i] = cnt[i//3] + 1
        pref[i+1] = pref[i] + cnt[i]


    for _ in range(inp()):
        solve(pref, cnt)

if __name__ == "__main__":
    main()



"""""""""
4
1 3
2 4
199999 200000
19 84
"""""""""

File: cf_cpp/cf_1999e.cpp
#include <iostream>
#include <vector>
#include <cmath>
using namespace std;

int inp() {
    int x;
    cin >> x;
    return x;
}

vector<int> inlt() {
    int n;
    vector<int> result;
    while (cin >> n) {
        result.push_back(n);
    }
    return result;
}

vector<char> insr() {
    string s;
    cin >> s;
    vector<char> result(s.begin(), s.end());
    return result;
}

vector<int> invr() {
    vector<int> result;
    int x;
    while (cin >> x) {
        result.push_back(x);
    }
    return result;
}

void solve() {
    const int N = 200007;
    int l, r;
    cin >> l >> r;
    
    vector<long long> pref(N, 0);
    vector<int> cnt(N, 0);
    
    for (int i = 1; i < N - 1; i++) {
        cnt[i] = cnt[i / 3] + 1;
        pref[i + 1] = pref[i] + cnt[i];
    }
    
    r += 1;
    long long sum = pref[r] - pref[l];
    int mn = cnt[l];
    
    cout << sum + mn << endl;
}

int main() {
    int t = inp();
    while (t--) {
        solve();
    }
    return 0;
}
